subprojects/core/src/main/java/org/gradle/internal/classpath/transforms/JarTransformFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.classpath.transforms;
+
+import org.gradle.internal.classpath.types.InstrumentingTypeRegistry;
+import org.gradle.internal.hash.Hasher;
+
+import java.io.File;
+
+/**
+ * Jar transform factory. There are some differences when instrumenting classes to be loaded by the instrumenting agent, this interface encapsulates them.
+ */
+public interface JarTransformFactory {
+    /**
+     * Modifies JAR content hash according to the algorithm implemented by this policy.
```

Comment:
:nail_care: this still uses the old `policy` name, maybe reword it like this:
```suggestion
     * Updates jar/directory content hash to account for the transform produced by this factory.
```